### PR TITLE
abstract

### PR DESCRIPTION
**Pull request by Farnazmdi**
In this branch, we address comments from Dr. Gross.

There are few comments that I have not  addressed yet and I appreciate your opinion about them:
1. Dr. Gross suggested changing the title to "A lineage tree-based hidden Markov model to quantify cellular heterogeneity and plasticity". My opinion is keeping the title as it is because heterogeneity is the result of plasticity. What do you think? 

For the other comments, please see below.

### CODE REVIEW

File: manuscript/03.results.md [hunk 1]
```diff
@@ -4,29 +4,29 @@
...
+After fitting, the model can provide a variety of information (Fig. {@fig:tree}c). First, it estimates the abundance of each state at time = 0 and the transition probabilities between these states which are the hyper-parameters of our tHMM model. Second, the distribution of cells' phenotypes in each state are estimated and can be compared to distinguish how cells of each state behave. The state of each individual cell can be predicted from the fit data or new measurements. Finally, the model provides a likelihood of each cell’s observations and therefore the data overall. This last quantity can be used, for example, to estimate the number of distinguishable cell states. When implementing these processes, we ensured that a cell’s measurements are defined through a modular interface, allowing other forms of data to be easily integrated.
```
Comment (aarmey) [inline manuscript/03.results.md:19]: Isn't "starting" more natural than "t=0"?

File: manuscript/03.results.md [hunk 2]
```diff
@@ -36,39 +36,39 @@ To incorporate this effect into our model, we marked cells or their specific cel
...
+Cells of even very distinct molecular states can have partly overlapping phenotypes due to non-heritable variation. Therefore, we sought to evaluate how different two states would need to be for us to accurately identify them as distinct (Fig. {@fig:wass}a). We varied the G1 phase duration of two states from identical (Wasserstein distance between the two G1 duration distributions zero) to very distinct (Wasserstein distance moret han 70) (Fig. {@fig:wass}b) and quantified the accuracy of our model. While the phenotypic observation of a given state had to be different for our model to accurately assign cells, even moderately overlapping phenotypes could be distinguished by using the lineage relationships of cells (Fig. {@fig:wass}c). As a baseline comparison, we used K-means clustering to cluster cells based on their phenotype without using their lineage relationships (Fig. {@fig:wass}c). A tHMM consistently out-performed this approach.  The model performance in censored and uncensored populations were similar (Fig. @fig:wass1, @fig:wass2). This shows that the heritability of a cell state can help to more accurately identify cells with partially overlapping phenotypes.
```
Comment (aarmey) [inline manuscript/03.results.md:47]: Typo in changes.

File: manuscript/01.abstract.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
...ticity. Here we apply a tree-based adaptation of a hidden Markov model (tHMM) that employs single cell lineages as input to learn the characteristic patterns of single cell heterogeneity and state transitions in an unsupervised fashion. This model enables single cell classification based on the phenotype of individual cells and their relatives for improved specificity in pinpointing the structure and dynamics of variability in drug response. Integrating this model with a modular interface for defining observed phenotypes allows the model to easily be adapted to any phenotype measured in single cells. To benchmark our model, we paired cell fate with either cell lifetimes or individual cell cycle phase lengths (G1 and S/G2) as our observed phenotypes on synthetic data and demonstrated that the model successfully classifies cells within experimentally-tractable dataset sizes. As an application, we analyzed experimental measurements of cell fate and phase duration in cancer cell populations treated with targeted therapies to determine the number of distinct subpopulations. In total, this tHMM framework allows for the flexible classification of single cell heterogeneity across lineages.
```
Comment (aarmey) [inline manuscript/01.abstract.md:3]: Gemcitabine is not a targeted therapy.

**Review by aarmey — CHANGES_REQUESTED** [review]
A few changes.

File: manuscript/03.results.md [hunk 4]
```diff
@@ -4,29 +4,29 @@
...
+Given the unique insights that single cell measurements on lineage trees can provide, we implemented a strategy for classifying cells based on their phenotype and lineage relationships. Given a set of measurements made across a lineage tree, we used a lineage tree-based hidden Markov model (tHMM) to fit these data (Fig. {@fig:tree}a). Like a typical hidden Markov model, a tHMM can infer the hidden discrete "states" of cells given a series of measurements where a state is defined by specific phenotype distributions. The inference of these states takes place using an iterative strategy wherein the states of each cell are predicted by the phenotype of both the cell and its relatives in a lineage ("expectation" step), and then each distribution of phenotypes is fit to match the cells within that state ("maximization" step) (Fig. {@fig:tree}b). This expectation-maximization (EM) process repeats until convergence.
```
[the remark this answers is not in the document]
Comment (aarmey) [reply manuscript/03.results.md:17]: I like adding an example to illustrate the point. I wouldn't say "resistance" is a phenotype itself. It's the lack of a phenotypic change in response to drug.

File: manuscript/01.introduction.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
+Large-scale profiling studies can find molecular features that associate with drug response using population-level samples [@PMID:31068700; @PMID:22460905]. These associations, while valuable, can miss the contribution of cell-to-cell heterogeneity, and especially stochastic changes in individual cell states that have significant effects on overall tumor drug response [@PMID:20371346; @PMID:21854987; @PMID:28607484]. The most common methods for quantifying drug response are metrics of tumor cell population expansion or contraction [@doi:10.1038/ni908; @pmid:8072198; @pmid:8655369; @pmid:20981102]. Recent research has made efforts to track phenotypic measurements of fitness at the single cell level [@pmid:29381859; @pmid:22886092]; however, even single cell measurements are typically performed with snapshots that subsequently miss the contribution of individual cells in the overall population response [@pmid:25421725].
 
 
 <!-- Lineage data is special -->
```
[the remark this answers is not in the document]
Comment (aarmey) [reply manuscript/01.introduction.md:10]: I think you could add something saying that, in population linkage analysis, genotype does change from individual. In cells, the genotype is mostly static but other things change. I make this point in my talks by explaining that cells are not people.

File: manuscript/01.abstract.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (aarmey) [reply manuscript/01.abstract.md:3]: I think plasticity refers to the adaptability of cell populations?

**Comment by aarmey** [comment]
> In this branch, we address comments from Dr. Gross.
> 
> There are few comments that I have not addressed yet and I appreciate your opinion about them:
> 
> 1. Dr. Gross suggested changing the title to "A lineage tree-based hidden Markov model to quantify cellular heterogeneity and plasticity". My opinion is keeping the title as it is because heterogeneity is the result of plasticity. What do you think?
> 
> For the other comments, please see below.

I'm neutral about the title. In the absence of an opinion, I'm inclined to change it.

File: manuscript/03.results.md (hunk 1 above)
Comment (Farnazmdi) [reply manuscript/03.results.md:19]: yes, but Dr. Gross asked the difference between starting abundance and the distribution, I thought this makes it a little bit more clear. I will change the sentence to emphasize that.

File: manuscript/01.introduction.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Farnazmdi) [reply manuscript/01.introduction.md:10]: I explained resistance as a state and cell death as a phenotype.

File: manuscript/03.results.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Farnazmdi) [reply manuscript/03.results.md:17]: I explained resistance as a state and cell death as a phenotype.

**Comment by Farnazmdi** [comment]
@aarmey This branch should be done now. I appreciate it if you could review the changes whenever convenient. Thank you.

**Pull request merged**